app/Http/Controllers/User/ProductController.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace App\Http\Controllers\User;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\ProductRequest;
+use App\Http\Requests\UpdateProductRequest;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Str;
+use Illuminate\Http\Request;
+
+class ProductController extends Controller
+{
+    public function add()
+    {
+        return view('User.Product.create_product');
+    }
+
+    public function store(ProductRequest $request)
+    {
+        if ($request->hasFile('image')) {
+            $image = $request->file('image');
+            $name = $image->getClientOriginalName();
+            $image->storeAs("uploads", $name, "public");
+        }
+
+        $product =
+        [
+            "name" => $request->name,
+            "description" => $request->description,
+            "images" => $name,
+            'slug' => Str::slug($request->name),
+            "price" => $request->price,
+        ];
+        $user = Auth::user();
```

Comment:
Vì `Auth::user` dùng nhiều nên hãy đưa vào thành 1 method `currentUser()` trong Controller chung.

Trong logic chỉ cần gọi `$this->currentUser()` là được.
Và cũng ko cần phải gán `$user = ` vì sau này sẽ khó đọc.